src/ucs/sys/sock.c

```diff
@@ -171,23 +171,25 @@ int ucs_netif_is_active(const char *if_name, sa_family_t af)
 
 unsigned ucs_netif_bond_ad_num_ports(const char *bond_name)
 {
-    long ad_num_ports = 1;
     ucs_status_t status;
+    long ad_num_ports;
     const char *bond_path;
     char lowest_dev_path_buf[PATH_MAX];
 
     status = ucs_topo_iface_lowest_device_path(bond_name, UCS_NETIF_DIR,
                                                lowest_dev_path_buf, PATH_MAX);
-    if (status == UCS_OK) {
-        bond_path = ucs_dirname(lowest_dev_path_buf, 1);
-        status    = ucs_read_file_number(&ad_num_ports, 1, "%s/%s", bond_path,
-                                         UCS_BOND_NUM_PORTS);
-        if ((status != UCS_OK) || (ad_num_ports <= 0) ||
-            (ad_num_ports > UINT_MAX)) {
-            ucs_trace("failed to read from %s/%s: %m, "
-                      "assuming 802.3ad bonding is disabled", bond_path,
-                      UCS_BOND_NUM_PORTS);
-        }
+    if (status != UCS_OK) {
+        return 1;
+    }
+
+    bond_path = ucs_dirname(lowest_dev_path_buf, 1);
+    status    = ucs_read_file_number(&ad_num_ports, 1, "%s/%s", bond_path,
+                                     UCS_BOND_NUM_PORTS_FILE);
+    if ((status != UCS_OK) || (ad_num_ports <= 0) ||
```

Comment:
There's no point in checking `(status != UCS_OK)` here.

Reply:
Hi @gleon99, per the implemation of `ucs_read_file_number()`, `ucs_read_file_number(...)` could return a value which's not `UCS_OK` and `ad_num_ports` is a random value because of `long ad_num_ports;`. So I think `(status != UCS_OK)` is needed. Thanks.